### ISSUE TITLE
should 'model_computed_fields' be a class attribute not an instance property?

### ISSUE THREAD

**Issue by hottwaj** [pydantic/pydantic#7986]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Hopefully title is quite self explanatory, but here is a bit of extra detail....

if I have a Model subclass "Schema", then I expected to be able to get to computed_fields via "model_computed_fields" attribute e.g. Schema.model_computed_fields, but this returns a "property" instance

looking under the hood, model_computed_fields returns data from `self.__pydantic_decorators__.computed_fields` - but `__pydantic_decorators__` is a class attribute and could presumably be accessed from the class without having a model instance

So in summary, this works:
`Schema.__pydantic_decorators__.computed_fields`

I expected this to work, but it doesn't:
`Schema.model_computed_fields`

Thanks!

### Python, Pydantic & OS Version

```Text
pydantic version: 2.2.0
        pydantic-core version: 2.6.0
          pydantic-core build: profile=release pgo=true mimalloc=true
               python version: 3.10.2 (main, Jan 28 2022, 16:19:08) [GCC 10.3.0]
                     platform: Linux-5.15.0-87-generic-x86_64-with-glibc2.31
     optional deps. installed: ['typing-extensions']
```

**Comment by sydney-runkle** [comment]
@hottwaj,

Thanks for reporting this! This seems like a reasonable change to make - I agree with your logic. Would you like to open a PR to make this change?

**Comment by hottwaj** [comment]
Great thanks!  I'm keen to submit a PR, but it might take a little time.  Will try to get to it soon :)

**Comment by geospackle** [comment]
@sydney-runkle  

could pick this up!

**Comment by Jocelyn-Gas** [comment]
@geospackle Hi, I don't know if you had the chance to have a look at this issue yet ?


The current implementation, defined in `main.py` is as follows:
```
@property
    def model_computed_fields(self) -> dict[str, ComputedFieldInfo]:
        """Get the computed fields of this model instance.

        Returns:
            A dictionary of computed field names and their corresponding `ComputedFieldInfo` objects.
        """
        return {k: v.info for k, v in self.__pydantic_decorators__.computed_fields.items()}
   ```
   
   In order to make this property to be both accessible as an instance property and a class property, I think it should be rewritten as such:
   ```
   @classmethod
   @property
    def model_computed_fields(cls) -> dict[str, ComputedFieldInfo]:
        """Get the computed fields of this model.

        Returns:
            A dictionary of computed field names and their corresponding `ComputedFieldInfo` objects.
        """
        return {k: v.info for k, v in cls.__pydantic_decorators__.computed_fields.items()}
     ```

Before opening a PR, I'd like to gather your thoughts @hottwaj @sydney-runkle @geospackle 🙏🏼

**Comment by hottwaj** [comment]
In principle that looks fine to me except that chaining classmethod and property together in that way is deprecated from python 3.11
https://docs.python.org/3.11/library/functions.html?highlight=classmethod#classmethod

So I think this needs to be set as a class level attribute during class creation instead

**Comment by geospackle** [comment]
Sounds good! I'll have a look after the holidays.
